### PR TITLE
feat: Add clear button to search bar

### PR DESCRIPTION
**Pull request by techwithanirudh**
This commit introduces a clear button to the SearchBar component using the X icon from Lucide-React. When the user enters a query in the input field, the clear button appears allowing them to easily remove the search term. The clear button is hidden when there is no search term entered.

### CODE REVIEW

**Comment by ClaraLeigh** [comment]
Hey, I love the enthusiasm but I'm not sure this feature will help our end User Experience.
 
1. This will create content shake, when the content is empty it will not have an icon, where as soon as you start typing the ux will change. In industry this is generally frowned upon these days and it is better to just make something look inactive.
2. I'm not sure we need this feature, everyone knows how to delete and with its position someone might click on it when going for the send button
3. On mobile you'd loose space and there is already a lack of space when trying to type things

File: client/src/components/Nav/NavLinks.jsx [hunk 1]
```diff
@@ -77,7 +77,7 @@ export default function NavLinks({ clearSearch, isSearchEnabled }) {
                 <Menu.Item as="div">
                   <NavLink
                     className={cn(
-                      'flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700',
+                      'flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700 rounded-none',
```
Comment (ClaraLeigh) [inline client/src/components/Nav/NavLinks.jsx:80]: do we need the rounded-none class here? it wasn't there previously

File: client/src/components/Nav/SearchBar.jsx [hunk 2]
```diff
@@ -37,8 +48,15 @@ const SearchBar = forwardRef((props, ref) => {
         placeholder="Search messages"
         onKeyUp={handleKeyUp}
       />
+      <X
+        className={`h-5 w-5 absolute right-3 cursor-pointer ${showClearIcon ? 'opacity-100' : 'opacity-0'} transition-opacity duration-1000`}
+        onClick={() => {
+          setSearchQuery('');
+          clearSearch();
```
Comment (ClaraLeigh) [inline client/src/components/Nav/SearchBar.jsx:55]: Do these lines do the same thing? also I'm not sure I can see a clearSearch() function included on this script anywhere?

File: client/src/components/Nav/NavLinks.jsx [hunk 3]
```diff
@@ -91,7 +91,7 @@ export default function NavLinks({ clearSearch, isSearchEnabled }) {
                 </Menu.Item>
                 <Menu.Item as="div">
                   <NavLink
-                    className="flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700"
+                    className="flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700 rounded-none"
```
Comment (ClaraLeigh) [inline client/src/components/Nav/NavLinks.jsx:94]: same again: do we need the rounded-none class here? it wasn't there previously

File: client/src/components/Input/SubmitButton.jsx [hunk 4]
```diff
@@ -88,7 +88,7 @@ export default function SubmitButton({
         disabled={disabled}
         className="group absolute bottom-0 right-0 flex h-[100%] w-[50px] items-center justify-center bg-transparent p-1 text-gray-500"
       >
-        <div className="m-1 mr-0 rounded-md p-2 pb-[10px] pt-[10px] group-hover:bg-gray-100 group-disabled:hover:bg-transparent dark:group-hover:bg-gray-900 dark:group-hover:text-gray-400 dark:group-disabled:hover:bg-transparent">
+        <div className="m-1 mr-0 rounded-md pt-[11px] pb-[9px] pl-[9.5px] pr-[7px] group-hover:bg-gray-100 group-disabled:hover:bg-transparent dark:group-hover:bg-gray-900 dark:group-hover:text-gray-400 dark:group-disabled:hover:bg-transparent">
```
Comment (ClaraLeigh) [inline client/src/components/Input/SubmitButton.jsx:91]: not sure I understand the changes to padding here? whats the reasoning?

**Review by ClaraLeigh — CHANGES_REQUESTED** [review]
Also I saw you posted the screenshot in discord. Might be good to pop it in here too so others can see the visual change. Before I saw that screenshot I actually thought we were talking about a different area

File: client/src/components/Nav/SearchBar.jsx (hunk 2 above)
Comment (techwithanirudh) [reply client/src/components/Nav/SearchBar.jsx:55]: I don't think so, dannya used it in the ```handleKeyUp``` function so I thought that was the best way to clear the input.

File: client/src/components/Nav/NavLinks.jsx (hunk 3 above)
Comment (techwithanirudh) [reply client/src/components/Nav/NavLinks.jsx:94]: I noticed that dannya made certain CSS modifications that unintentionally affected the consistency of a few menu items. To address this issue, I took the initiative to make the necessary adjustments and restore their visual coherence.

File: client/src/components/Nav/NavLinks.jsx (hunk 1 above)
Comment (techwithanirudh) [reply client/src/components/Nav/NavLinks.jsx:80]: I noticed that dannya made certain CSS modifications that unintentionally affected the consistency of a few menu items. To address this issue, I took the initiative to make the necessary adjustments and restore their visual coherence.

File: client/src/components/Input/SubmitButton.jsx (hunk 4 above)
Comment (techwithanirudh) [reply client/src/components/Input/SubmitButton.jsx:91]: The submit button had incorrect padding, so I resolved this issue. This pull request encompasses an overall CSS improvement.

File: client/src/components/Nav/SearchBar.jsx (hunk 2 above)
Comment (ClaraLeigh) [reply client/src/components/Nav/SearchBar.jsx:55]: ah I see the function included above now, just wanted to check it was linking somewhere

**Review by ClaraLeigh — APPROVED** [review]
Lgtm. Assuming the rounded-none css is needed but @danny-avila would know for sure as you mention its a bug from their css changes. Same with the extra submit button padding

**Review by hubbdavid — APPROVED** [review]

**Review by danny-avila — APPROVED** [review]

**Pull request merged**